conan/internal/util/files.py

```diff
@@ -256,15 +261,44 @@ def mkdir(path):
     os.makedirs(path)
 
 
-def tar_extract(fileobj, destination_dir):
+def tar_extract(fileobj, destination_dir, compression_plugin=None, conf=None):
+    if compression_plugin:
+        _tar_extract_with_plugin(fileobj, destination_dir, compression_plugin, conf)
+        return
     the_tar = tarfile.open(fileobj=fileobj)
     # NOTE: The errorlevel=2 has been removed because it was failing in Win10, it didn't allow to
     # "could not change modification time", with time=0
     # the_tar.errorlevel = 2  # raise exception if any error
     the_tar.extraction_filter = (lambda member, path: member)  # fully_trusted, avoid Py3.14 break
     the_tar.extractall(path=destination_dir)
     the_tar.close()
-
+    if list(Path(destination_dir).glob(f"{COMPRESSED_PLUGIN_TAR_NAME}.*")):
+        raise ConanException(f"Error while extracting {os.path.basename(fileobj.name)}.\n"
+                             "This file has been compressed using a `compression` plugin.\n"
+                             "If your organization uses this plugin, ensure it is correctly installed on your environment.")
+
+def _tar_extract_with_plugin(fileobj, destination_dir, compression_plugin, conf):
+    """First remove tar.gz wrapper and then call the plugin to extract"""
+    with tempfile.TemporaryDirectory() as temp_dir:
```

Comment:
Conan doesn't really use system TemporaryDirectories for anything, it tries to do everything in the Conan cache, in cache folders, so if something goes wrong the files are there, and also there is no "leakage" of pacakges, recipes or code in anywhere else beside the Conan cache, that can be easily wiped.

Reply:
Yes, I understand the concern.
Let me think about it.